### PR TITLE
Separate rewards pools for DKG and heartbeats

### PR DESCRIPTION
**Pull request by lukasz-zimnoch**
Refs: #2725
Depends on: #2729

This pull request gets rid of the "maintenance pool" concept and introduces two separate rewards pools instead. The `dkgRewardsPool` is funded with relay request fees or with external donates. Those funds are used to cover rewards for DKG result approvers and DKG timeout notifiers. The second `heartbeatNotifierRewardsPool` is funded only with external donates and is used to cover actors notifying about the failed group heartbeat.

### CODE REVIEW

**Comment by lukasz-zimnoch** [comment]
Ready for review. Keeping it draft until #2729 lands on main.

File: solidity/random-beacon/contracts/RandomBeacon.sol [hunk 1]
```diff
@@ -122,17 +122,34 @@ contract RandomBeacon is Ownable {
     ///         operator affected.
     uint256 public dkgMaliciousResultNotificationRewardMultiplier;
 
+    // External dependencies
```
Comment (pdyraga) [inline solidity/random-beacon/contracts/RandomBeacon.sol:125]: These comments separating fields 👌🏻 💯

File: solidity/random-beacon/test/RandomBeacon.GroupCreation.test.ts [hunk 2]
```diff
@@ -4,6 +4,7 @@ import { ethers, waffle, helpers, getUnnamedAccounts } from "hardhat"
 import { expect } from "chai"
 import type { BigNumber, ContractTransaction, Signer } from "ethers"
 import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers"
+import { describe } from "mocha"
```
Comment (pdyraga) [inline solidity/random-beacon/test/RandomBeacon.GroupCreation.test.ts:7]: 🤔 

Is it needeed?

**Review by pdyraga — COMMENTED** [review]
Just one comment about import, everything else looks great.

File: solidity/random-beacon/test/RandomBeacon.GroupCreation.test.ts (hunk 2 above)
Comment (lukasz-zimnoch) [reply solidity/random-beacon/test/RandomBeacon.GroupCreation.test.ts]: https://github.com/keep-network/keep-core/pull/2735/commits/7cf5c0e2a88586386d58fbc84d1c50b5f87d3919

**Review by pdyraga — APPROVED** [review]

**Pull request merged**